src/main/java/com/ss/lms/entity/Copies.java

```diff
@@ -2,52 +2,77 @@
 
 import java.io.Serializable;
 
-public class Copies implements Serializable {
+import javax.persistence.Column;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
 
-	private static final long serialVersionUID = -5459027020967892222L;
-	private Integer noOfCopies;
-	private Integer bookId;
-	private Integer branchId;
-	private Book book;
-	private Branch libraryBranch;
+import org.springframework.stereotype.Component;
 
-	public Book getBook() {
-		return book;
-	}
+import com.fasterxml.jackson.annotation.JsonIgnore;
 
-	public void setBook(Book book) {
-		this.book = book;
-	}
+@Entity
+@Table(name = "tbl_book_copies")
+@Component
+public class Copies implements Serializable {
 
-	public Branch getLibraryBranch() {
-		return libraryBranch;
-	}
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 6483994900241520391L;
 
-	public void setLibraryBranch(Branch branch) {
-		this.libraryBranch = branch;
-	}
+	@EmbeddedId
+	private CopiesId copiesId;
+
+	@Column(name = "noOfCopies")
+	private Long noOfCopies;
+	
+	@JsonIgnore
+	@ManyToOne(fetch = FetchType.LAZY)
+	@JoinColumn(name = "bookId", insertable = false, updatable = false)
+	private Book book;
+	
+	@JsonIgnore
```

Comment:
same here, unless your specifically displaying only copies from a branch wouldnt you not want @jsonIgnore here. Yall already using @jsonignore on the other side of the realtionship so there wouldnt be any problems with looping